test/memory.test.js

```diff
@@ -76,6 +85,35 @@ describe('Memory connector', function() {
       }, done);
     });
 
+    it('should not have out of sequence read/write', function(done) {
```

Comment:
That was not clear at all. In general, it's better to write self-contained tests, any side-effects should be triggered by the test code.

At minimum, please add a comment explaining why it is important to keep this particular test in the current position (i.e. what you wrote here on GH).